### PR TITLE
create api for generate short link

### PR DESCRIPTION
**Pull request by samiullah111299**

### CODE REVIEW

File: src/main/java/com/example/quily/handler/ShortLinkHandler.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+@Component
+public class ShortLinkHandler {
+
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/handler/ShortLinkHandler.java:15]: remove extra line

File: src/main/java/com/example/quily/handler/ShortLinkHandler.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @Autowired
+    ShortLinkServices shortLinkServices;
+    public Mono<ServerResponse> createShortLink(ServerRequest serverRequest) {
+
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/handler/ShortLinkHandler.java:19]: remove extra line

File: src/main/java/com/example/quily/model/ShortLink.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    private String userID;
+    private String originalLink;
+    private String shortedLink;
+    private String creationDate;
+    private String expiryDate;
+    private boolean active;
+
+    public ShortLink(){}
+
+    public ShortLink( String originalLink, String shortedLink, String creationDate, String expiryDate) {
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/model/ShortLink.java:21]:
```suggestion
    public ShortLink(String originalLink, String shortedLink, String creationDate, String expiryDate) {
```

File: src/main/java/com/example/quily/repositories/ShortLinkRepository.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.quily.repositories;
+
+import com.example.quily.model.ShortLink;
+import org.springframework.data.r2dbc.repository.R2dbcRepository;
+
+public interface ShortLinkRepository extends R2dbcRepository<ShortLink, ShortLink> {
+}
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/repositories/ShortLinkRepository.java:7]: repository takes first datatype of model and second data type of primary key
```suggestion
public interface ShortLinkRepository extends R2dbcRepository<ShortLink, Long> {
}
```

File: src/main/java/com/example/quily/request/CreateShortLinkRequest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.time.LocalDateTime;
+
+public class CreateShortLinkRequest {
+    private String userID;
+    private String originalLink;
+    private String expiryDate;
+    private String creationDate;
+
+    public CreateShortLinkRequest(){}
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/request/CreateShortLinkRequest.java:11]: remove all constructer and setter from here.

File: src/main/java/com/example/quily/response/KGSResponse.java [hunk 6]
```diff
@@ -25,4 +25,8 @@ public String getHashKey() {
     public void setHashKey(String hashKey) {
         this.hashKey = hashKey;
     }
+
+    public String getShortenedLink() {
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/response/KGSResponse.java:29]: remove it

File: src/main/java/com/example/quily/response/ShortLinkResponse.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    public ShortLinkResponse(String userID, String originalLink, String shortedLink, String expiryDate, String creationDate) {
+        this.userID = userID;
+        this.originalLink = originalLink;
+        this.shortedLink = shortedLink;
+        this.expiryDate = expiryDate;
+        this.creationDate = creationDate;
+    }
+
+    public String getUserID() {
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/response/ShortLinkResponse.java:18]: remove all getter setter from here

File: src/main/java/com/example/quily/router/ShortLinkRouter.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.reactive.function.server.ServerResponse;
+
+@Configuration
+public class ShortLinkRouter {
+    @Autowired
+    private ShortLinkHandler shortLinkHandler;
+
+    @Bean
+    public RouterFunction<ServerResponse> orderRoutes() {
+
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/router/ShortLinkRouter.java:21]: remove extra line

File: src/main/java/com/example/quily/services/ShortLinkServices.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import com.example.quily.response.ShortLinkResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+import java.time.LocalDateTime;
+
+@Service
+public class ShortLinkServices {
+
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/services/ShortLinkServices.java:15]: remove extra line

File: src/main/java/com/example/quily/services/ShortLinkServices.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    @Value("${ShortedLink}")
+    String ShortedLinks;
+
+    @Autowired
+    ShortLinkRepository shortLinkRepository;
+    ShortLinkResponse shortLinkResponse;
+    KGSService kgsService;
+
+    public ShortLinkServices(KGSService kgsService) {
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/services/ShortLinkServices.java:24]: remove contructer

File: src/main/java/com/example/quily/services/ShortLinkServices.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+                                LocalDateTime.now().toString(),
+                                createShortLinkRequest.getExpiryDate()));
+    }
+
+
+    public Mono<ShortLink> saveShortLink(Mono<ShortLink> shortLink) {
+        return shortLink.flatMap(shortLinkRepository::save);
+    }
+    public ShortLinkResponse mapShortLinkToShortLinkResponse(ShortLink shortLink) {
+        return new ShortLinkResponse(shortLink.getUserID(),shortLink.getOriginalLink(),shortLink.getShortedLink(),shortLink.getExpiryDate(),shortLink.getCreationDate());
```
Comment (Nisaar123) [inline src/main/java/com/example/quily/services/ShortLinkServices.java:43]:
```suggestion
        return new ShortLinkResponse(shortLink.getUserID(), shortLink.getOriginalLink(), shortLink.getShortedLink(), shortLink.getExpiryDate(), shortLink.getCreationDate());
```

File: src/main/resources/application.properties [hunk 12]
```diff
@@ -6,4 +6,5 @@ spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQLDialect
 spring.jpa.show-sql=true
 spring.jpa.generate-ddl=true
 spring.jpa.hibernate.ddl-auto=update
-server.port=8081
\ No newline at end of file
+server.port=8081
+ShortedLink = https://quily/
```
Comment (Nisaar123) [inline src/main/resources/application.properties:10]: put it in util

**Review by Nisaar123 — COMMENTED** [review]
resolve comment

**Comment by samiullah111299** [comment]
Comment resolve

**Comment by samiullah111299** [comment]
comment resolve

**Review by Nisaar123 — APPROVED** [review]
LGTM

**Pull request merged**